### PR TITLE
Fix reward updates

### LINKED ISSUE #2121: Wrong reward is displayed in the pool info
I see strange behavior for stake pools, there are pretty large rewards in the blobber pool
[https://demo1.zus.network/sharder01/v1/screst/6dba10422e368813802877a85039d3985d96760[…]d1337d20f74d579c04ce579455f4126219e2d2812385&provider_type=3](https://demo1.zus.network/sharder01/v1/screst/6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7/getStakePoolStat?provider_id=7572413f2e9112215480d1337d20f74d579c04ce579455f4126219e2d2812385&provider_type=3)
```json
"id": "1746b06bb09f55ee01b33b5e2e055d6cc7a900cb57c0a3a5eaabb8a0e7745802",
"balance": 10000000000,
"delegate_id": "1746b06bb09f55ee01b33b5e2e055d6cc7a900cb57c0a3a5eaabb8a0e7745802",
"rewards": 27098703609650,
"unstake": false,
"provider_id": "",
"provider_type": 0,
"total_reward": 29028476163016,
"total_penalty": 0,
"status": "active",
"round_created": 2878698
```
but when claiming rewards there are collected few
```json
"transaction_output": "{\"ID\":0,\"CreatedAt\":\"0001-01-01T00:00:00Z\",\"UpdatedAt\":\"0001-01-01T00:00:00Z\",\"amount\":3,\"block_number\":2880455,\"client_id\":\"1746b06bb09f55ee01b33b5e2e055d6cc7a900cb57c0a3a5eaabb8a0e7745802\",\"pool_id\":\"\",\"provider_type\":\"3\",\"provider_id\":\"7572413f2e9112215480d1337d20f74d579c04ce579455f4126219e2d2812385\"}",
```
the same value is in db in mint_rewards table
looks like delegate_pools.rewards is not displayed correctly and there should be much smaller value, did you face anything similar?

### PR DESCRIPTION
**Pull request by Sriep**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test: https://github.com/0chain/system_test/pull/555
- zboxcli:
- zwalletcli:
- Other: ...
- This has https://github.com/0chain/0chain/pull/2100 and bits of https://github.com/0chain/0chain/pull/2059 merged in. It is needed to test https://github.com/0chain/system_test/pull/555 = miner and sharder fee system tests.

### CODE REVIEW

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go [hunk 1]
```diff
@@ -13,60 +14,31 @@ import (
...
-				PoolID:               k,
-				Reward:               v,
-				TotalReward:          v,
-				TotalPenalty:         spus[i].DelegatePenalties[k],
-				RoundPoolLastUpdated: round,
-			})
-		}
-
-		// append remaining penalties if any
-		for k, v := range spus[i].DelegatePenalties {
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/dbs/event/stakepool.go:53]: Should we ignore this `DelegatePenalities`?

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/dbs/event/stakepool.go (hunk 1 above)
Comment (Sriep) [reply code/go/0chain.net/smartcontract/dbs/event/stakepool.go:53]: I don't know. We should not be using delegate penalties, but they might still be used somewhere; in this case, we should fix that bug.

The problem with delegate penalties is that they can easily result in negative rewards. Negative rewards will never be collected, just forgotten, and can be used to bypass paying fines. 

Slashes for failed challenges should come from the stake pool, not the rewards. There should be enough in the stake pools to cover the slashes.

https://github.com/0chain/0chain/issues/2122

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | update_rewards  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by peterlimg** [comment]
@Sriep please resolve the conflicts

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | miner_fee_rewards  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | update_rewards  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | true  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | update_rewards  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix-rounding-off-for-balance  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**